browser/ui/views/frame/split_view/brave_multi_contents_view.cc

```diff
@@ -79,12 +90,39 @@ void BraveMultiContentsView::Layout(PassKey) {
   contents_container_views_[1]->SetBoundsRect(end_rect);
 }
 
+void BraveMultiContentsView::SetActiveIndex(int index) {
+  MultiContentsView::SetActiveIndex(index);
+
+  UpdateSecondaryLocationBar();
+}
+
 float BraveMultiContentsView::GetCornerRadius() const {
   return BraveBrowser::ShouldUseBraveWebViewRoundedCorners(
              browser_view_->browser())
              ? BraveContentsViewUtil::kBorderRadius + kBorderThickness
              : 0;
 }
 
+void BraveMultiContentsView::UpdateSecondaryLocationBar() {
+  int inactive_index = active_index_ == 0 ? 1 : 0;
+  if (!secondary_location_bar_) {
+    secondary_location_bar_ = std::make_unique<SplitViewLocationBar>(
+        browser_view_->browser()->profile()->GetPrefs(),
+        contents_container_views_[inactive_index]);
+    secondary_location_bar_widget_ = std::make_unique<views::Widget>();
+
+    secondary_location_bar_widget_->Init(
+        SplitViewLocationBar::GetWidgetInitParams(
+            GetWidget()->GetNativeView(), secondary_location_bar_.get()));
+  }
+
+  // Inactive web contents/view should be set to secondary location bar
+  // as it's attached to inactive contents view.
+  secondary_location_bar_->SetWebContents(
+      GetInactiveContentsView()->GetWebContents());
+  secondary_location_bar_->SetParentWebView(
+      contents_container_views_[inactive_index]);
```

Comment:
could these lines be in an `else` block for the above `if`? Otherwise I think we set the parent web view twice (once in ctor, once here) when we construct the location bar.

Reply:
Alternatively, we could just not pass it in to the constructor? Doesn't seem like its that necessary.